src/meta/src/manager/catalog/fragment.rs

```diff
@@ -853,21 +920,41 @@ impl FragmentManager {
         Ok(())
     }
 
-    pub async fn all_worker_parallel_units(&self) -> HashMap<WorkerId, HashSet<ParallelUnitId>> {
-        let mut all_worker_parallel_units = HashMap::new();
+    pub async fn all_worker_slots(&self) -> HashMap<WorkerId, HashSet<WorkerSlotId>> {
+        let mut all_worker_slots = HashMap::new();
         let map = &self.core.read().await.table_fragments;
+
+        let mut worker_fragment_map = HashMap::new();
+
         for table_fragment in map.values() {
-            table_fragment.worker_parallel_units().into_iter().for_each(
-                |(worker_id, parallel_units)| {
-                    all_worker_parallel_units
-                        .entry(worker_id)
-                        .or_insert_with(HashSet::new)
-                        .extend(parallel_units);
-                },
+            for fragment in table_fragment.fragments() {
+                for actor in fragment.get_actors() {
+                    let worker = table_fragment
+                        .actor_status
+                        .get(&actor.actor_id)
+                        .unwrap()
+                        .get_parallel_unit()
+                        .unwrap()
+                        .get_worker_node_id();
+
+                    let fragment_ref = worker_fragment_map.entry(worker).or_insert(HashMap::new());
```

Comment:
Nits: seems like we can only maintain a `HashMap<WorkerId, usize>` directly to store worker's max fragment parallelism and update it by grouping the `actor_status`.